tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -239,17 +239,6 @@ public:
 /// Classes with dictionaries that can be inspected by TClass
 template <typename T, typename = void>
 class RField final : public RClassField {
-protected:
-   void ConstructValue(void *where) const final
-   {
-      if constexpr (std::is_default_constructible_v<T>) {
-         new (where) T();
-      } else {
-         // If there is no default constructor, try with the IO constructor
-         new (where) T(static_cast<TRootIOCtor *>(nullptr));
```

Comment:
Since the tests still work, I assume `TClass::New()` takes proper care of using the I/O constructor where applicable...?

Reply:
Yes, in fact now I learned that `TClass::New()` allows more customization from the user: https://root.cern.ch/doc/master/classTClass.html#a1e952a88f08eb6ec665a7ac862860689. I cross-checked that removing the IO constructor of `class IOConstructor` in `tree/ntuple/v7/test/CustomStruct.hxx` fails the `ntuple-types` test, so it seems to be working with this PR.

Reply:
Yes, this is essentially the whole point of `TClass::New`: construct the object during I/O operations :)